### PR TITLE
DOCS: Merge FAQ into docs on multi-dimensional arrays

### PR DESCRIPTION
**Pull request by ChrisChinchilla**
### Checklist
- [x] Code compiles correctly
- [x] All tests are passing
- [x] New tests have been created which fail without the change (if possible)
- [x] README / documentation was extended, if necessary
- [x] Changelog entry (if change is visible to the user)
- [x] Used meaningful commit messages

### Description

This PR moves a FAQ question on multi-dimensional arrays into docs.

### CODE REVIEW

File: docs/control-structures.rst [hunk 1]
```diff
@@ -29,8 +29,13 @@ something like::
         }
     }
 
-Input parameters can be used just as any other local variable
-can be used, they can also be assigned to.
+Input parameters can be used as any other local variable and they can also be assigned to.
```
Comment (ChrisChinchilla) [inline docs/control-structures.rst]: @chriseth I'd like to clarify this, but need to check something. Does it mean they can have values assigned?

Comment (axic) [reply docs/control-structures.rst]: Yes they can be assigned to.

File: docs/control-structures.rst [hunk 2]
```diff
@@ -29,8 +29,13 @@ something like::
...
+
+.. note::
+    A function cannot accept a multi-dimensional array as an input parameter.
```
Comment (axic) [inline docs/control-structures.rst]: This is missing one important keyword which was present in the FAQ: "non-internal function".

Basically a function marked internal supports it without any tweaks.  Functions which are external or public only support it if ABIEncoderV2 is enabled.

File: docs/control-structures.rst [hunk 3]
```diff
@@ -29,8 +29,13 @@ something like::
...
+    This functionality is possible if you enable the new experimental ``ABIEncoderV2``
+    feature by adding ``pragma experimental ABIEncoderV2;`` to your source file.
+    This feature introduces other experimental changes that you can read more about in the :ref:`ABI` guide.
```
Comment (axic) [inline docs/control-structures.rst]: I don't think the "ABI" section explains what features ABIEncoderV2 introduces.

**Comment by axic** [comment]
Need to also be rebased.

File: docs/control-structures.rst (hunk 3 above)
Comment (ChrisChinchilla) [reply docs/control-structures.rst]: @axic In that case, I'm not aware of anywhere in the docs that does. Is there anywhere else I can find the content to get it included?

**Comment by ChrisChinchilla** [comment]
@axic rebase was messy, hopefully this works.

File: docs/control-structures.rst (hunk 3 above)
Comment (erak) [reply docs/control-structures.rst]: Unfortunately, there is no section that covers the ABIEncV2 exclusively. We could remove the last sentence for now, create that subsection with another PR and then link to that section from all the places where the pragma is used.

**Comment by erak** [comment]
@ChrisChinchilla It still has conflicts and some tests are failing. Do you want to give it another try? :)

File: docs/control-structures.rst [hunk 4]
```diff
@@ -53,12 +65,40 @@ write::
...
 
-The names of output parameters can be omitted.
-The output values can also be specified using ``return`` statements,
-which are also capable of :ref:`returning multiple values<multi-return>`.
-Return parameters can be used as any other local variable and they
-are zero-initialized; if they are not explicitly
-set, they stay zero.
+The names of return variables can be omitted.
+Return variables can be used as any other local variable and they
+are zero-initialized. If they are not explicitly
```
Comment (erak) [inline docs/control-structures.rst]:
```suggestion
are initialized with their :ref:`default value <default-value>` and have that value unless explicitly set.
```
I'm not 100% sure about the exact wording, though.

File: docs/control-structures.rst (hunk 3 above)
Comment (ChrisChinchilla) [reply docs/control-structures.rst]: @bit-shift 👍

File: docs/control-structures.rst (hunk 4 above)
Comment (ChrisChinchilla) [reply docs/control-structures.rst]: @bit-shift Ah yes, this was added elsewhere too 👍

**Comment by ChrisChinchilla** [comment]
@bit-shift At least now there are no merge conflicts, had to rebase like 5 times. Weird :/

**Comment by erak** [comment]
Wait, I think you've lost the commit that removed the content from the FAQ

**Comment by lazaridiscom** [comment]
(Me, yesterday, fix merge-conflicts&rebase : 5 min. codework, 2 hours struggle with git/github. This pain will never end i guess)

**Comment by ChrisChinchilla** [comment]
Good catch @bit-shift think this is done now.

**Comment by ChrisChinchilla** [comment]
Rebased and squashed

**Review by chriseth — APPROVED** [review]

**Pull request merged**